charts/pipeline/templates/secrets.yaml

```diff
@@ -16,25 +16,36 @@ data:
 
 ---
 
-{{ if .Values.service.tls  }}
+{{ if not (and .Values.service.tls.secretName .Values.service.tls.caSecretName) }}
 {{- $cn := include "pipeline.fullname" . -}}
 {{- $externalDNS := print $cn "." .Release.Namespace -}}
 {{- $ca := genCA "pipeline-ca" 365 -}}
 {{- $cert := genSignedCert $cn (list "127.0.0.1") (list $cn $externalDNS) 365 $ca -}}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "pipeline.fullname" . }}-tls
+  name: {{ include "pipeline-tls-secret-name" . }}
   labels:
     app: {{ include "pipeline.name" . }}
     chart: {{ include "pipeline.chart" . }}
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
 data:
-  ca.crt: {{ $ca.Cert | b64enc }}
-  ca.key: {{ $ca.Key | b64enc }}
-  server.crt: {{ $cert.Cert | b64enc }}
-  server.key: {{ $cert.Key | b64enc }}
+  tls.crt: {{ $cert.Cert | b64enc }}
+  tls.key: {{ $cert.Key | b64enc }}
+---
+apiVersion: v1
```

Comment:
Cert manager places CA in the same secret, under `ca.crt` key (if we are talking about cert manager friendly setup). Also, I'm not sure from a consumer point of view we need them in separate secrets.

We only need the CA key to actually sign certs, but that will rarely happen inside the cluster (unless we use the self-signed signer of cert manager, but even then the CA cert is placed in the final secret as well).

Reply:
Fixed.